src/lib.rs

```diff
@@ -110,8 +112,19 @@ pub const XOR_NAME_LEN: usize = 32;
 pub struct XorName(pub [u8; XOR_NAME_LEN]);
 
 impl XorName {
+    /// Generate a XorName for the given content (for content-addressable-storage)
+    pub fn from_content(content_parts: &[&[u8]]) -> Self {
+        let mut sha3 = Sha3::v256();
+        for part in content_parts.iter() {
```

Comment:
Minor: you don't need to call `iter()`. Slices already implement `IntoIterator`.